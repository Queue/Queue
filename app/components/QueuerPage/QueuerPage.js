//
// QueuerPage is a higher level component
// for displaying editable features of a single queuer

import React, { Component } from 'react';
import {
  View,
  Text,
  StyleSheet,
  TextInput
} from 'react-native';

import Common from '../../lib/common';
import Colors from '../../lib/colors';

export default class QueuerPage extends Component {
  constructor(props) {
    super(props);

    this.state = {
      nameText: this.props.queuer.name,
      partySizeText: this.props.queuer.partySize,
      phoneNumberText: this.props.queuer.phoneNumber,
      notesText: ''
    }
  }

  componentDidMount() {
    //Common.logLess(this.props.queuer);
  }

  render() {
    return (
      <View style={styles.container}>
        {Object.keys(this.props.queuer).length ? (
          <View>
            <Text style={styles.header}>
              {this.props.queuer.name}
            </Text>
            <View style={{marginTop: 20}}>
              <View style={{marginTop: 10, marginBottom: 10}}>
                <Text>Name</Text>
                <TextInput
                  style={{paddingLeft: 10, height: 40, borderColor: Colors.primaryBackground, borderWidth: 1}}
                  onChangeText={(text) => {this.setState({nameText: text})}}
                  placeholder={this.props.queuer.name}
                  value={this.state.nameText}
                />
              </View>
              <View style={{marginTop: 10, marginBottom: 10}}>
                <Text>Party Size</Text>
                <TextInput
                  style={{paddingLeft: 10, height: 40, borderColor: Colors.primaryBackground, borderWidth: 1}}
                  onChangeText={(text) => {this.setState({partySizeText: text})}}
                  placeholder={this.props.queuer.partySize}
                  value={this.state.partySizeText}
                />
              </View>
              <View style={{marginTop: 10, marginBottom: 10}}>
                <Text>Phone Number</Text>
                <TextInput
                  style={{paddingLeft: 10, height: 40, borderColor: Colors.primaryBackground, borderWidth: 1}}
                  onChangeText={(text) => {this.setState({phoneNumberText: text})}}
                  placeholder={this.props.queuer.phoneNumber}
                  value={this.state.phoneNumberText}
                />
              </View>
              <View style={{marginTop: 10, marginBottom: 10}}>
                <Text>Notes</Text>
                <TextInput
                  multiline={true}
                  numberOfLines={4}
                  style={{paddingLeft: 10, fontSize: 17, height: 100, borderColor: Colors.primaryBackground, borderWidth: 1}}
                  onChangeText={(text) => {this.setState({notesText: text})}}
                  placeholder={this.props.queuer.notes}
                  value={this.state.notesText}
                />
              </View>
            </View>
          </View>
        ) : (
          <View style={{
            flexDirection: 'column',
            justifyContent: 'center',
            alignItems: 'center',
            marginTop: 250,
            maxWidth: 400}}>
              <Text style={{fontSize: 30, color: Colors.primaryBackground}}>
                Select a person in queue to edit their values.
              </Text>
          </View>
        )}
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    padding: 13
  },
  header: {
    fontSize: 55,
    fontWeight: '100',
    letterSpacing: 2
  }
});
