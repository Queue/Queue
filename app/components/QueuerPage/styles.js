//
// QueuerPage Styles

import { StyleSheet } from 'react-native';
import Colors from '../../lib/colors';

export default styles = StyleSheet.create({
  container: {
    padding: 13
  },
  header: {
    fontSize: 55,
    fontWeight: '100',
    letterSpacing: 2
  },
  buttonText: {
    textAlign: 'center',
    color: 'white',
    fontSize: 20,
    padding: 15,
    letterSpacing: 1
  },
  col: {
    margin: 5
  },
  label: {
    letterSpacing: 1,
    marginBottom: 5,
    color: 'grey'
  },
  unselectedContainer: {
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
    marginTop: 250,
    maxWidth: 400
  },
  unselectedText: {
    fontSize: 30,
    color: Colors.primaryBackground
  },
  inputContainer: {
    marginTop: 10,
    marginBottom: 10
  },
  input: {
    paddingLeft: 15,
    height: 45,
    borderColor: Colors.primaryBackground,
    borderWidth: 1
  },
  textareaInput: {
    paddingTop: 7,
    paddingLeft: 15,
    fontSize: 17,
    height: 140,
    borderColor: Colors.primaryBackground,
    borderWidth: 1
  }
});
