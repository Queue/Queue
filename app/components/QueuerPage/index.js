//
// QueuerPage Index

import QueuerPage from './QueuerPage';
export default QueuerPage;
