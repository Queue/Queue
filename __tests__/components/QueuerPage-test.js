//
// QueuerPage component test

import React from 'react';
import QueuerPage from '../../app/components/QueuerPage';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <QueuerPage
      queuer={{name: 'fred'}}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
