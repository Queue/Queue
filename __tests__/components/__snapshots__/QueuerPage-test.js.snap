// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "padding": 13,
      "width": "100%",
    }
  }
>
  <RCTScrollView
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutoAutomaticScroll={true}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    keyboardDismissMode="interactive"
    onScroll={[Function]}
    scrollEventThrottle={0}
    showsVerticalScrollIndicator={true}
  >
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "fontFamily": "Open Sans",
            "fontSize": 55,
            "fontWeight": "100",
            "letterSpacing": 2,
          }
        }
      >
        &lt;Name&gt;
      </Text>
      <View
        style={
          Object {
            "marginTop": 20,
          }
        }
      >
        <View
          style={
            Object {
              "borderBottomWidth": 1,
              "borderColor": "#eaeaea",
              "marginBottom": 8,
              "marginTop": 8,
              "width": "100%",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "grey",
                "fontFamily": "Open Sans",
                "letterSpacing": 1,
                "marginBottom": 2,
              }
            }
          >
            Name
          </Text>
          <TextInput
            label="Name"
            onChangeText={undefined}
            style={
              Object {
                "fontFamily": "Open Sans",
                "height": 40,
                "paddingLeft": 15,
              }
            }
            type="text"
            value={undefined}
          />
        </View>
        <View
          style={
            Object {
              "borderBottomWidth": 1,
              "borderColor": "#eaeaea",
              "marginBottom": 8,
              "marginTop": 8,
              "width": "100%",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "grey",
                "fontFamily": "Open Sans",
                "letterSpacing": 1,
                "marginBottom": 2,
              }
            }
          >
            Party Size
          </Text>
          <TextInput
            label="Party Size"
            onChangeText={undefined}
            style={
              Object {
                "fontFamily": "Open Sans",
                "height": 40,
                "paddingLeft": 15,
              }
            }
            type="text"
            value={undefined}
          />
        </View>
        <View
          style={
            Object {
              "borderBottomWidth": 1,
              "borderColor": "#eaeaea",
              "marginBottom": 8,
              "marginTop": 8,
              "width": "100%",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "grey",
                "fontFamily": "Open Sans",
                "letterSpacing": 1,
                "marginBottom": 2,
              }
            }
          >
            Phone Number
          </Text>
          <TextInput
            keyboardType="phone-pad"
            label="Phone Number"
            onChangeText={undefined}
            placeholder="Phone Number"
            style={
              Array [
                Object {
                  "fontFamily": "Open Sans",
                  "height": 40,
                  "paddingLeft": 15,
                },
                undefined,
              ]
            }
            type="number"
            value={undefined}
          />
        </View>
        <View
          style={
            Object {
              "borderBottomWidth": 1,
              "borderColor": "#eaeaea",
              "marginBottom": 8,
              "marginTop": 8,
              "width": "100%",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "grey",
                "fontFamily": "Open Sans",
                "letterSpacing": 1,
                "marginBottom": 2,
              }
            }
          >
            Notes
          </Text>
          <TextInput
            label="Notes"
            multiline={true}
            numberOfLines={4}
            onChangeText={undefined}
            placeholder="Notes"
            style={
              Array [
                Object {
                  "borderColor": "#eaeaea",
                  "borderLeftWidth": 1,
                  "fontFamily": "Open Sans",
                  "fontSize": 17,
                  "height": 170,
                  "paddingLeft": 15,
                  "paddingTop": 7,
                },
                undefined,
              ]
            }
            type="textarea"
            value={undefined}
          />
        </View>
      </View>
      <View
        style={
          Object {
            "marginTop": 25,
          }
        }
      >
        <View
          style={
            Array [
              undefined,
              Object {
                "alignContent": undefined,
                "alignItems": "flex-start",
                "flexDirection": "row",
                "flexWrap": "nowrap",
                "height": undefined,
                "justifyContent": "flex-start",
              },
            ]
          }
        >
          <View
            onLayout={[Function]}
            style={
              Array [
                undefined,
                Object {
                  "alignItems": "flex-start",
                  "flexDirection": "column",
                  "justifyContent": "flex-start",
                  "marginLeft": 0,
                  "marginRight": 0,
                  "width": "100%",
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "width": "100%",
                }
              }
            >
              <View
                accessibilityComponentType="button"
                accessibilityLabel={undefined}
                accessibilityTraits="button"
                accessible={true}
                collapsable={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                nativeID={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "borderColor": "#8FD6D6",
                    "borderRadius": 4,
                    "borderWidth": 1,
                    "marginBottom": 5,
                    "marginLeft": 5,
                    "marginRight": 5,
                    "marginTop": 5,
                    "opacity": 1,
                    "overflow": "hidden",
                    "padding": 8,
                  }
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "#007aff",
                        "fontSize": 17,
                        "fontWeight": "500",
                        "textAlign": "center",
                      },
                      null,
                      Array [
                        Object {
                          "fontFamily": "Open Sans",
                          "fontWeight": "500",
                        },
                        Object {
                          "color": "#8FD6D6",
                          "fontSize": 20,
                        },
                      ],
                      null,
                    ]
                  }
                >
                  Text
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Array [
              undefined,
              Object {
                "alignContent": undefined,
                "alignItems": "flex-start",
                "flexDirection": "row",
                "flexWrap": "nowrap",
                "height": undefined,
                "justifyContent": "flex-start",
              },
            ]
          }
        >
          <View
            onLayout={[Function]}
            style={
              Array [
                undefined,
                Object {
                  "alignItems": "flex-start",
                  "flexDirection": "column",
                  "justifyContent": "flex-start",
                  "marginLeft": 0,
                  "marginRight": 0,
                  "width": "50%",
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "width": "100%",
                }
              }
            >
              <View
                accessibilityComponentType="button"
                accessibilityLabel={undefined}
                accessibilityTraits="button"
                accessible={true}
                collapsable={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                nativeID={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "borderColor": "#6CDA6C",
                    "borderRadius": 4,
                    "borderWidth": 1,
                    "marginBottom": 5,
                    "marginLeft": 5,
                    "marginRight": 5,
                    "marginTop": 5,
                    "opacity": 1,
                    "overflow": "hidden",
                    "padding": 8,
                  }
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "#007aff",
                        "fontSize": 17,
                        "fontWeight": "500",
                        "textAlign": "center",
                      },
                      null,
                      Array [
                        Object {
                          "fontFamily": "Open Sans",
                          "fontWeight": "500",
                        },
                        Object {
                          "color": "#6CDA6C",
                          "fontSize": 20,
                        },
                      ],
                      null,
                    ]
                  }
                >
                  Seat
                </Text>
              </View>
            </View>
          </View>
          <View
            onLayout={[Function]}
            style={
              Array [
                undefined,
                Object {
                  "alignItems": "flex-start",
                  "flexDirection": "column",
                  "justifyContent": "flex-start",
                  "marginLeft": 0,
                  "marginRight": 0,
                  "width": "50%",
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "width": "100%",
                }
              }
            >
              <View
                accessibilityComponentType="button"
                accessibilityLabel={undefined}
                accessibilityTraits="button"
                accessible={true}
                collapsable={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                nativeID={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "borderColor": "#FB5A5A",
                    "borderRadius": 4,
                    "borderWidth": 1,
                    "marginBottom": 5,
                    "marginLeft": 5,
                    "marginRight": 5,
                    "marginTop": 5,
                    "opacity": 1,
                    "overflow": "hidden",
                    "padding": 8,
                  }
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "#007aff",
                        "fontSize": 17,
                        "fontWeight": "500",
                        "textAlign": "center",
                      },
                      null,
                      Array [
                        Object {
                          "fontFamily": "Open Sans",
                          "fontWeight": "500",
                        },
                        Object {
                          "color": "#FB5A5A",
                          "fontSize": 20,
                        },
                      ],
                      null,
                    ]
                  }
                >
                  Remove
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
